sympy/solvers/solveset.py

```diff
@@ -234,6 +231,84 @@ def _invert_complex(f, g_ys, symbol):
     return (f, g_ys)
 
 
+def _invert_abs(f, g_ys, symbol):
+    """Helper function for inverting absolute value functions.
+
+    Returns the complete result of inverting an absolute value function.
+    Alongwith finding all the possible solutions of the given function,
+    the routine determines a set of conditions which must be satified
+    for their existence.
+
+    The returned values are determined by the condition
+    If it is certain that all these conditions are met, a `FiniteSet` of all
+    possible solutions is returned.
+    If it can be concluded that atleast one condition cannot be satified, an
+    `EmptySet` is returned.
+    In all other cases, a `ConditionSet` of the solutions, with all the required
+    conditions specified, is returned.
+
+    """
+
+    n = Dummy('n', real=True)
+
+    conditions = S.EmptySet
+    condition_interval = Interval(0, oo)
+
+    if f.is_positive:
+        f_n = f
+    else:
+        f_n = f.args[0]
+    g_x, values = _invert_real(f_n, Union(imageset(Lambda(n, n), g_ys),
```

Comment:
+1.
I wrote this because of two reasons: 

- To clearly show the logic: Inverting the `Union` of both the positive and negative images of `g_ys` under the transformation `f_n`.
- For code readability